### DISCUSSION TITLE
What are the components and what are the example applications?

### DISCUSSION THREAD

**Discussion by tomcrane** [digirati-co-uk/iiif-canvas-panel#20] [General]
(not a _how might we_ ... need to do some work on this list)

## Components used to build the demonstrations and tools

 - Library code (vault etc)
 - A canvas rendering component (the thing called canvas panel)
 - A textual annotation list (for text lines, AV transcripts)
 - ...

## Tools and applications

 - A simple viewer
 - A content state creator widget (see #18) - this uses the simple viewer in a modal popup
 - A simple viewer with text (a bit like that implied by https://canvas-panel.digirati.com/developer-stories/viewer1.html)
 - A simple 2-up paged object viewer

## Demonstrations

- #1 Static Image (simple and complex canvases - the code is the same)
- #2 Responsive Image (simple and complex scenes)
- #4 Deep Zoom
- #5 Constrained text (demonstrate a variety of text highlighting mechanisms
- #7 a - show what the output code would look like
- #7 b - demonstrate a pseudo CMS in which an editor can create some data (content state + text) that can be used with that output code in a template.
- #9 A simple viewer showing a canvas with multiple different images on it (not choice)
- #10 A canvas with multispectral images (maybe add to one of the simple viewers above)
- #12 Show different ways in which highlights, boxes and shapes can be drawn on the canvas programatically
- #13 Show different ways existing linked annotations, and supplied at "code time" annotations, can be rendered - links, tags, etc
- #15 a Simple viewer with text to the side (simulate Mirador example)
- #15 b Simple viewer with selectable text overlay
- #16 A demonstrator that logs everything the user is doing on the canvas to show event handling
- #17 2-up canvas composed to 2 existing single page canvas rendered by simple viewer
- A simple "strip" viewer ("As a continuous scroll with the images side-by-side, and a long thumbnail navigator area below")
- As a slide show with forward and backward arrows to cycle through, and switch a box of thumbnails as a navigation aid below

**Comment by stephenwf** [comment]
Additional demonstration to supplement selectable text overlays may be to provide a similar selecting ability for illustrations or photographs. You could enable interactions where right-clicking would work like a normal `<img />` tag and allow you to save it as if it was an image as part of a page.
